src/customer-service/Actions/CreateIncidentAction.cs

```diff
@@ -0,0 +1,158 @@
+﻿using customer_service.Factories;
+using customer_service.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace customer_service.Actions
+{
+    public class CreateIncidentAction
+    {
+        /**
+         * Class: CreateIncidentAction
+         * Purpose: Prompts the user for information about a new incident and saves that incident to the database
+         * Author: Garrett Vangilder
+         * Methods:
+         *     static void ReadInput() - Prompts for incident info and saves to DB
+         */
+        public CreateIncidentAction()
+        {
+
+        }
+        /**
+        * Purpose: Reads user input for new incident information
+        * Return:
+        *     void
+        */
+        public static void ReadInput()
+        {
+            IncidentFactory incidentFactory = new IncidentFactory();
+            CustomerFactory customerFactory = new CustomerFactory();
+            OrderFactory orderFactory = new OrderFactory();
+            IncidentTypeFactory incidentTypeFactory = new IncidentTypeFactory();
+            List<IncidentType> incidentTypeList = incidentTypeFactory.getAll();
+            List<Customer> customerList = customerFactory.getAll();
+            List<Incident> incidents = incidentFactory.getAll();
+            Incident incident = new Incident();
+            incident.DateCreated = DateTime.Now;
+
+            Console.WriteLine("Enter the customers first and last name to start:");
+            Console.WriteLine("***********************************************************************");
+
+
+            foreach (Customer customer in customerList)
+            {
+                Console.WriteLine($"{customer.CustomerId} { customer.FirstName} {customer.LastName} ");
+            }
+            Console.WriteLine("***********************************************************************");
+
+            string customerName = Console.ReadLine();
+            int customerId = customerFactory.getCustomerByFullName(customerName).CustomerId;

```

Comment:
You can use linq to query your `customerList` enumerable for a matching FirstName and LastName. Load that into a while loop. No need to hit the database again